openvdb/util/CpuTimer.h

```diff
@@ -56,65 +57,130 @@ namespace util {
+    /// @brief Prints message and start timer.
     ///
     /// @note Should normally be followed by a call to stop()
-    CpuTimer(const std::string& msg) { this->start(msg); }
+    CpuTimer(const std::string& msg, std::ostream& os = std::cerr) : mOutStream(os), mT0() { this->start(msg); }
 
     /// @brief Start timer.
     ///
-    /// @note Should normally be followed by a call to time()
+    /// @note Should normally be followed by a call to milliseconds() or stop(std::string)
     inline void start() { mT0 = tbb::tick_count::now(); }
 
-    /// @brief Print message and re-start timer.
+    /// @brief Print message and start timer.
     ///
     /// @note Should normally be followed by a call to stop()
     inline void start(const std::string& msg)
     {
-        std::cerr << msg << " ... ";
+        mOutStream << msg << " ...";
         this->start();
     }
 
-    /// @brief Stop previous timer, print message and re-start timer.
+    /// @brief Return Time difference in milliseconds since construction or start was called.
     ///
-    /// @note Should normally be followed by a call to stop()
-    inline void restart(const std::string& msg)
+    /// @note Combine this method with start() to get timing without any outputs.
+    inline double milliseconds() const
     {
-        this->stop();
-        this->start(msg);
+        tbb::tick_count::interval_t dt = tbb::tick_count::now() - mT0;
+        return 1000.0*dt.seconds();
     }
 
-    /// Return Time diference in milliseconds since construction or start was called.
-    inline double delta() const
+    /// @brief This method is identical to milliseconds() - will soon be deprecated
+    inline double delta() const { return this->milliseconds(); }
+
+    inline std::string time() const
     {
-        tbb::tick_count::interval_t dt = tbb::tick_count::now() - mT0;
-        return 1000.0*dt.seconds();
+        const double msec = this->milliseconds();
+        std::ostringstream os;
+        printTime(os, msec, "", "", true, 4, 1);
+        return os.str();
     }
 
-    /// @brief Print time in milliseconds since construction or start was called.
-    inline void stop() const
+    /// @brief Returns and prints time in milliseconds since construction or start was called.
+    ///
+    /// @note Combine this method with start(std::string) to print at start and stop of task being timed.
+    inline double stop() const
     {
-        const double t = this->delta();
-        std::ostringstream ostr;
-        ostr << "completed in " << std::setprecision(3) << t << " ms\n";
-        std::cerr << ostr.str();
+        const double msec = this->milliseconds();
+        printTime(mOutStream, msec, " completed in ", "\n", true, 4, 1);
+        return msec;
     }
 
-private:
+    /// @brief Returns and prints time in milliseconds since construction or start was called.
+    ///
+    /// @note Combine this method with start() to delay output to will completion of task being timed.
```

Comment:
typo (though I'm not sure what this is intended to be) - "delay output to will completion of task being timed"